File: src/java/exceptions.java
public class exceptions {

    private static void DoStuff(boolean inError) throws Exception {
        if (inError) {
            throw new Exception("An error occured!");
        }

        System.out.println("Doing stuff!");
    }

    public static void main(String[] args) {
        try {
            DoStuff(false);
            DoStuff(true);
            DoStuff(false);
        } catch (Exception ex) {
            System.out.println("ERRO: " + ex.getMessage());
        }
    }

}

File: src/scala/exceptions.scala
object Exceptions extends App {
    def doStuff(inError: Boolean) {
        if (inError)
            throw new Exception("An error occured!")
        
        println("Doing stuff!")
    }

    try {
        doStuff(false)
        doStuff(true)
        doStuff(false)

    } catch {
         case e: Exception => println("ERROR: " + e.getMessage())
    }
}